### ISSUE TITLE
Pyreverse unable to detect relative imports

### ISSUE THREAD

**Issue by balint-k** [PyCQA/pylint#9559]
### Bug description

```python
The pyreverse doesn't detect any relative import (from . import something), doesn't show any connection between packages.
```

### Command used

```shell
pyreverse something -o html
```

### Pylint output

```python
The html files, but in the packages result, there were no connections between relative imported files
```

### Expected behavior

A html, where i find connection between the relative imported files

### Pylint version

```shell
pylint 3.1.0
astroid 3.1.0
python 3.9.12.
```

**Comment by balintkeri** [comment]
[dummy.zip](https://github.com/pylint-dev/pylint/files/15108283/dummy.zip)

I have this example package. If i run pyreverse dummy -o html, i got this package diagram:

![image](https://github.com/pylint-dev/pylint/assets/52506432/77db7bd3-78cd-4e56-b6ae-645f5a96b50c)

The import arrow is missing between the dummy2 and dummy3, dummy4

**Comment by balintkeri** [comment]
@DanielNoord
